pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/Environment.groovy

```diff
@@ -266,4 +253,11 @@ public class Environment implements Serializable {
         public boolean isLiteral
         public Object value
     }
+
+    private String resolveAsScript(Binding binding, String script) {
+        String toRun = Utils.prepareForEvalToString(script)
+        SecureGroovyScript toExec = new SecureGroovyScript(toRun, true)
+            .configuring(ApprovalContext.create().withCurrentUser().withKey(Utils.getCurrentJobName()))
```

Comment:
`withKey`: _Associates a unique key with this approval. If not null, any previous approval of the same kind with the same key will be canceled and replaced by this one._

Since you seem to be calling this several times in the same script with the same key they would overwrite eachother. Is that intended?